sycl/doc/extensions/DiscardQueueEvents/SYCL_EXT_ONEAPI_DISCARD_QUEUE_EVENTS.asciidoc

```diff
@@ -0,0 +1,223 @@
+get_info<info::event::reference_count>() const;
+
+// must throw an exception with the errc::invalid_object error code.
+template <info::event_profiling param>
+typename param::return_type get_profiling_info() const;
+----
+
+The behavior when passing _invalid_ event to handler API:
+[source,c++]
+----
+// must throw an exception with the errc::invalid_object error code.
+handler::depends_on(event Event)
+
+// must throw an exception with the errc::invalid_object error code.
+handler::depends_on(const std::vector<event> &Events)
+----
+
+== Limitations
+
+The `discard_queue_events` property should be taken as a hint for SYCL implementation.
+In all kernel cases, SYCL implementation returns _invalid_ event, regardless of whether the optimization was done.
+Optimization occurs if the queue has `discard_queue_events` property and all features presented below are not met:
+
+ - Using out-of-order queue or using together with `enable_profiling` property
+ - Using https://github.com/intel/llvm/blob/sycl/sycl/doc/extensions/Assert/SYCL_ONEAPI_ASSERT.asciidoc[fallback assert feature]
+ - Using streams, buffer/image accessors (excluding local accessors)
+
+
+See details for each below:
+
+=== Using out-of-order queue and/or using together with enable_profiling property
+
+No optimization if a queue is created with the `discard_queue_events` property and
+the property list includes the `enable_profiling` property or does not include `in_order` property.
+
+=== Using fallback assert feature
+
+If a kernel that uses the fallback assert feature is submitted to a queue created with
+the `discard_queue_events` property, the implementation must throw an exception with the errc::invalid_object error code.
+It is important to disable the logic for detecting assertion failures in kernels in principle
+since it is not enough just not to use the `assert` macro inside kernel code or compile with `NDEBUG`.
+To disable the logic, developer must compile with SYCL_DISABLE_FALLBACK_ASSERT macro defined,
+for more details see https://github.com/intel/llvm/blob/sycl/sycl/doc/PreprocessorMacros.md[document]
+
+[source,c++]
+----
+  sycl::property_list props{ext::oneapi::property::queue::discard_queue_events{},
+                            property::queue::in_order()};
+  sycl::queue Queue( props );
+  // if the submission of kernel1 is compiled with the logic for detecting assertion failures enabled,
+  // then the submission of the kernel throws an exception with the errc::invalid_object error code.
+  Queue.parallel_for<class kernel1>(NDRange, [=](nd_item<1> item){ do_smth(); });
+----
+
+=== Using streams, buffer/image accessors (excluding local accessors)
+
+No optimization if a kernel that uses stream objects, buffer or image accessors is submitted to a queue created with
+the `discard_queue_events` property. But using local accessors does not affect optimization.
+
+*NOTE : This extension is only compatible with kernels using Unified Shared Memory (USM) pointers.

```

Comment:
Not sure that with latest updates it's still true.

Reply:
ok, removed the note